flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliUtils.java

```diff
@@ -109,7 +109,7 @@ public static boolean createFile(final Path filePath) {
             if (parent == null) {
                 return false;
             }
-            Files.createDirectories(parent);
+            Files.createDirectories(parent.toRealPath());
```

Comment:
`toRealPath()` is a function that only works on files that already exist.

Reply:
Thanks for you review, fixed the bug and added related test case.